Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -75,8 +98,36 @@ public override void Serialize(IIonWriter writer, object item)
         {
             options.TypeAnnotator.Apply(options, writer, targetType);
             writer.StepIn(IonType.Struct);
+
+            var serializedWithGetter = new List<string>();
+            
+            // Serialize the values returned from IonPropertyGetter annotated getter methods.
+            foreach (var method in targetType.GetMethods())
+            {
+                var getMethod = (IonPropertyGetter)method.GetCustomAttribute(typeof(IonPropertyGetter));
+                
+                // A getter method should have zero parameters.
+                if (getMethod?.IonPropertyName == null || method.GetParameters().Length != 0)
+                {
+                    continue;
+                }
+
+                writer.SetFieldName(getMethod.IonPropertyName);
+                var getValue = method.Invoke(item, Array.Empty<object>());
+                ionSerializer.Serialize(writer, getValue);
+                
+                serializedWithGetter.Add(getMethod.IonPropertyName);
+            }
+
+            // Serialize any properties that satisfy the options/attributes.
             foreach (var property in targetType.GetProperties())
             {
+                if (serializedWithGetter.Contains(property.Name))
```

Comment:
Can we clarify intent here? What is the behaviour if you have a method, field, and property that are all annotated with the same name? Currently you're choosing to ignore the property and field, but that seems unintuitive.

Reply:
Clarified with Ian and we agreed to skip duplicates if they've already been serialized via a method.